include/nanobind/nb_cast.h

```diff
@@ -606,6 +606,19 @@ object cast(T &&value, rv_policy policy = rv_policy::automatic_reference) {
     return steal(h);
 }
 
+template <typename T>
+object cast(T &&value, handle parent) {
+    detail::cleanup_list cleanup(parent.ptr());
+    handle h = detail::make_caster<T>::from_cpp((detail::forward_t<T>) value,
+                                                rv_policy::reference_internal, &cleanup);
+    if (!h.is_valid())
+        detail::raise_cast_error();
+
+    cleanup.release();
```

Comment:
I think you will need to do the `.release()` call one line above already. Otherwise we might raise an exception with leaks.

Reply:
Can you also add this additional function to the `docs/api_core.rst` file?